CLI/support/sparkle_job_help.py

```diff
@@ -125,17 +128,75 @@ def get_running_jobs() -> list[SlurmRun]:
             if run.status == Status.WAITING or run.status == Status.RUNNING]
 
 
+def get_dependencies(jobs: list[SlurmRun]) -> list[SlurmRun]:
+    """Adds Dependencies. Should be removed once RunRunner provides this feature."""
+    for job in jobs:
+        with job.json_filepath.open() as file:
+            data = json.load(file)
+        if len(data["dependencies"]) > 0:
+            job_ids = [dep["run_id"] for dep in data["dependencies"]]
+        else:
+            job_ids = []
+        job.dependencies = job_ids
+    return jobs
+
+
+def clear_console(lines: int) -> None:
+    """Clears console for updated table to be printed."""
+    # \033 is the escape character (ESC) in ASCII
+    # [{lines}A is the escape sequence that moves the cursor up.
+    print(f"\033[{lines}A", end="")
+    # [J is the exape sequence that clears the console from the cursor down
+    print("\033[J", end="")
+
+
 def wait_for_all_jobs() -> None:
     """Wait for all active jobs to finish executing."""
-    running_jobs = get_running_jobs()
-    prev_jobs = len(running_jobs) + 1
-    while len(running_jobs) > 0:
-        if len(running_jobs) < prev_jobs:
-            print(f"Waiting for {len(running_jobs)} jobs...", flush=True)
-        time.sleep(10.0)
-        prev_jobs = len(running_jobs)
-        running_jobs = [run for run in running_jobs
-                        if run.status == Status.WAITING or run.status == Status.RUNNING]
+    jobs = get_running_jobs()
+    verbosity_setting = gv.settings.get_output_verbosity()
+    running_jobs = [run for run in jobs
+                    if run.status == Status.WAITING or run.status == Status.RUNNING]
+
+    # If verbosity is quiet there is no need for further information
+    if verbosity_setting == VerbosityLevel.QUIET:
+        prev_jobs = len(running_jobs) + 1
+        while len(running_jobs) > 0:
+            if len(running_jobs) < prev_jobs:
+                print(f"Waiting for {len(running_jobs)} jobs...", flush=True)
+            time.sleep(3.0)
+            prev_jobs = len(running_jobs)
+            running_jobs = [run for run in running_jobs
+                            if run.status == Status.WAITING
+                            or run.status == Status.RUNNING]
+
+    # If verbosity is standard the command will print a table with relevant information
+    elif verbosity_setting == VerbosityLevel.STANDARD:
+        # Collect dependencies and partitions for each job
+        jobs = get_dependencies(jobs)
+        while len(running_jobs) > 0:
+            # Information to be printed to the table
+            information = [["RunId", "Name", "Status", "Dependencies", "Finished Jobs"]]
+            running_jobs = [run for run in running_jobs
```

Comment:
Can we detect change here? E.g. if we are not going to be producing a different table, it would be better to skip towards the sleep()

Reply:
We could check by comparing the previous running_jobs with the new object and then also compare the finished_jobs_count before and after. However that is basically all the computing we need anyway for the table, so we're not really getting anything out of that check